test/extended/router/http2.go

```diff
@@ -54,29 +62,86 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
+		http2RouterShardConfigPath = exutil.FixturePath("testdata", "router", "router-shard.yaml")
+		http2SourceBackendPath     = exutil.FixturePath("testdata", "router", "router-http2-server.backend")
+
+		oc = exutil.NewCLI("router-http2")
+
+		shardConfigPath string // computed
 	)
 
 	// this hook must be registered before the framework namespace teardown
 	// hook
 	g.AfterEach(func() {
 		if g.CurrentGinkgoTestDescription().Failed {
-			client := routeclientset.NewForConfigOrDie(oc.AdminConfig()).RouteV1().Routes(oc.KubeFramework().Namespace.Name)
-			if routes, _ := client.List(context.Background(), metav1.ListOptions{}); routes != nil {
-				outputIngress(routes.Items...)
-			}
 			exutil.DumpPodLogsStartingWith("http2", oc)
+			exutil.DumpPodLogsStartingWithInNamespace("router", "openshift-ingress", oc.AsAdmin())
+		}
+		if len(shardConfigPath) > 0 {
+			if err := oc.AsAdmin().Run("delete").Args("-n", "openshift-ingress-operator", "-f", shardConfigPath).Execute(); err != nil {
+				e2e.Logf("deleting ingress controller failed: %v\n", err)
+			}
 		}
 	})
 
 	g.Describe("The HAProxy router", func() {
 		g.It("should pass the http2 tests", func() {
-			g.Skip("disabled for https://bugzilla.redhat.com/show_bug.cgi?id=1853711")
-			g.By(fmt.Sprintf("creating test fixture from a config file %q", configPath))
-			err := oc.Run("new-app").Args("-f", configPath).Execute()
+			infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "failed to get cluster-wide infrastructure")
+			switch infra.Status.PlatformStatus.Type {
+			case configv1.OvirtPlatformType, configv1.KubevirtPlatformType, configv1.LibvirtPlatformType, configv1.VSpherePlatformType:
+				g.Skip("Skip on platforms where the default router is not exposed by a load balancer service.")
+			}
+
+			defaultDomain, err := getDefaultIngressClusterDomainName(oc, 5*time.Minute)
+			o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
+
+			srcTarGz, err := makeCompressedTarArchive([]string{http2SourceBackendPath})
 			o.Expect(err).NotTo(o.HaveOccurred())
-			e2e.ExpectNoError(e2epod.WaitForPodRunningInNamespaceSlow(oc.KubeClient(), "http2", oc.KubeFramework().Namespace.Name))
+			base64SrcTarGz := strings.Join(split(base64.StdEncoding.EncodeToString(srcTarGz), 76), "\n")
+
+			g.By(fmt.Sprintf("creating service from a config file %q", http2ServiceConfigPath))
+			err = oc.Run("new-app").Args("-f", http2ServiceConfigPath, "-p", "BASE64_SRC_TGZ="+base64SrcTarGz).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			e2e.ExpectNoError(e2epod.WaitForPodRunningInNamespaceSlow(oc.KubeClient(), "http2", oc.Namespace()), "http2 backend server pod not running")
+
+			// certificate start and end time are very
+			// lenient to avoid any clock drift between
+			// between the test machine and the cluster
```

Comment:
nit - there are 2 "between" s

Reply:
Done in https://github.com/openshift/origin/pull/25897/commits/b84944f1116f690213418d6893be1c7353749fa4